build-system/tasks/visual-diff/index.js

```diff
@@ -469,59 +471,59 @@ async function snapshotWebpages(browser, webpages) {
       // since Puppeteer doesn't always understand Chrome's network activity, so
       // ignore timeouts again.
       const pagePromise = (async () => {
-        const responseWatcher = new Promise((resolve, reject) => {
-          const responseTimeout = setTimeout(() => {
-            reject(
-              new puppeteer.TimeoutError(
-                `Response was not received in test ${testName} for page ` +
-                  `${webpage.url} after ${NAVIGATE_TIMEOUT_MS}ms`
-              )
-            );
-          }, NAVIGATE_TIMEOUT_MS);
+        try {
```

Comment:
@danielrozenberg Since I'm guessing you've worked with this code a decent bit, would you mind looking at this function/file to double-check the changes?

Reply:
`try { try {` is usually a code-smell and makes it a lot harder to follow the flow of the program. Let's rewrite this as:
```javascript
try {
  const responseWatcher = new Promise((resolve, reject) => {
    ...
  });

  log('verbose', 'Navigating to page', ...
  await Promise.all(...
  log('verbose', 'Page navigation of test', ...
} catch (navigationError) {
  hasWarnings = true;
  addTestError(testErrors, name, ...
  log('warning', 'Continuing to verify page regardless...');
}

// --> completely outside of the first try clause

try {
  await percySnapshot(page, name, snapshotOptions);
} catch (testError) {
  addTestError(testErrors, name, ...
  await percySnapshot(page, name, SNAPSHOT_SINGLE_BUILD_OPTIONS);
}

// --> instead of a "finally" clause, since execution will get here regardless of previous failures now

log(hasWarnings ? 'warning' : 'info', ...
```